colossus/src/main/scala/colossus/core/server/Server.scala

```diff
@@ -84,7 +84,9 @@ private[colossus] class Server(io: IOSystem, serverConfig: ServerConfig, stateRe
   }
 
   val ss: ServerSocket = ssc.socket()
-  val address          = new InetSocketAddress(settings.port)
+  val addresses: Seq[InetSocketAddress] =
+    serverConfig.settings.addresses.map(address => new InetSocketAddress(address, settings.port)) // "localhost" default
```

Comment:
Is `InetSocketAddress("localhost", settings.port)` the same as `InetSocketAddress(settings.port)`. Seemed a little more complicated under the hood?

Reply:
I'm not sure we actually want to default to localhost.  `new InetSocketAddress(settings.port)` defaults to binding with the "wildcard" address, aka `0.0.0.0`, but usually localhost resolves to `127.0.0.1`.  So I'm pretty sure defaulting to localhost means that it'll only accept connections coming from the same machine.

Though on the other hand, defaulting to bind to localhost is a much more secure default I suppose.  But we'd have to be _extra_ clear about it in all docs, especially since it would be changing default behavior and could cause mass panic as unsuspecting devs roll out a "version upgrade" that suddenly stops accepting all incoming connections.

Reply:
I think keeping the wildcard address is probably sensible for now.

Reply:
agreed that sticking to the wildcard address will lead to a better version-upgrade experience. 

interesting point on the security aspect of using localhost .. possibly something to consider when doing a bigger version bump that includes a migration guide?